### PR TITLE
[READY]Da small cytology update. Banana glutton and proper clown mob reagents.

### PR DESCRIPTION
**Pull request by Krysonism**
## About The Pull Request
This PR reworks the relatively uninteresting clown blob mob into a new type of mob: the banana glutton.

The banana glutton is able to eat anything edible as well as small mobs. Some items that are too funny to digest(cream pies, tomatoes and meat clowns) are stored in a special throat pouch when ingested, and can regurgitated at high speed for comedic effect.

Eating bananas has an additional benefit as they heal the glutton based on potency and the peel is retained in the pouch for later use.

Lastly, you can ride the glutton if sentient or tamed and it is impervious to small chip damage due to a force threshold of 10.

The old clown blob was one of the mobs that would spawn when growing a clown cell line failed, since the banana glutton is a higher tier mob it is now a proper cell line and the flesh clown is now takes its place amongst the failure state mobs.

Sprites:

I have given the pugnacious gnome clown blob sprite a few layers of visual polish to make it more pleasing(or repulsive) to look at.

clown_glutton_4_comp
clown_blob_4

Clown cell lines reagent fix:

I have assigned the clown cell lines appropriate required, supplemental and suppressive reagents  so the xenobiologist can select for the one they want to grow.

## Why It's Good For The Game
Since their introduction the growing of clown mobs has been quite popular. Sadly they are quite mechanically lacking and could be a lot more interesting.

The banana glutton has enough going on to be attractive to ghosts and its active ability can be both used for light hearted pranking or serious combat.

The new mob is more dependent on cooperation from humans(botanists in particular) to be effective in combat as compared to previous high tier mobs such as the gelatinous cube.

## Changelog
:cl:
add: Added a new cytology clown mob, the banana glutton!
del: Removed the old clown blob mob.
fix: The cytology clown mob cell lines now have properly assigned reagents. 
imageadd: The old clown blob sprite has been partially resprited.
/:cl:

### CODE REVIEW

**Comment by IndieanaJones** [comment]
130 health isn't much of anything when considering the fact that the mob takes full damage from all sources, and it has decent speed for being as massive as it is.  Blobbernauts, for reference, have 200 health but take half damage from brute, all while being faster and easier to produce.  While these mobs can heal, it only really helps if you stockpile bananas somewhere for it to pull from, and chances are having a human use sucture on you is simply more effective.  My suggestion is to give it some melee resistance to make it compare to other tanky options, or simply buff its health pool.

Also, why does this mob take double damage from cloning?  Cloning damage doesn't do anything to simple mobs.

**Comment by Tlaltecuhtli** [comment]
it seems to have a long tongue? why  not give it a tentacle attack like ling has?

**Comment by Potato-Masher** [comment]
To help with effective health, Blunt objects outta have a high chance of harmlessly bouncing off it's girth.

**Comment by Krysonism** [comment]
> 
> 
> 130 health isn't much of anything when considering the fact that the mob takes full damage from all sources and is a very easy sprite to click on.[...]

 Good point, it ought to have higher effective health since it is so easy to click.

> Also, I don't think upping the melee damage would be bad either, and I would also suggest the basic attack be a bite. Do you see the size of those teeth? I wouldn't want to go anywhere near it.

Maybe. I would at least consider it. It would make less sense if we want to retain the bike horn attack sound.

>Might want to reconsider eating small mobs considering vore and all that.

It is not vore, it is a convenient bite sized snack. I think it might be funny to eat Ian or some bothersome swarmers or something.

> Finally, why does this mob take double damage from cloning? Cloning damage doesn't do anything to simple mobs.

I was under the impression simple mobs could take damage from cloning sources. I might have been mistaken, I will test tomorrow.  The idea was to make several big cytology mobs weak to the decloner so that it would have a niche in countering mob spam.

**Review by Twaticus — APPROVED** [review]

File: code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm [hunk 1]
```diff
@@ -299,21 +299,152 @@
...
+	riding.set_vehicle_dir_layer(EAST, ABOVE_MOB_LAYER)
+	riding.set_vehicle_dir_layer(WEST, ABOVE_MOB_LAYER)
+
+	//these 0 offsets are needed for the riding component to work properly with base pixel offset mobs for some cursed reason.
+	riding.set_vehicle_dir_offsets(SOUTH, 0, 0)
+	riding.set_vehicle_dir_offsets(NORTH, 0, 0)
+	riding.set_vehicle_dir_offsets(EAST, 0, 0)
+	riding.set_vehicle_dir_offsets(WEST, 0, 0)
+
+	riding.drive_verb = "ride"
```
Comment (LemonInTheDark) [inline code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:397]: your code does not compile, please resolve this

Comment (Krysonism) [reply code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm]: I did

File: code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm [hunk 2]
```diff
@@ -299,21 +299,149 @@
...
+		prank_pouch += eaten_atom
+
+	else if(istype(eaten_atom, /obj/item/food/grown/banana))
+		var/obj/item/food/grown/banana/banana_morsel = eaten_atom
+		adjustBruteLoss(-banana_morsel.seed.potency * 0.25)
+		prank_pouch += banana_morsel.generate_trash(src)
+		qdel(eaten_atom)
+	else
+		qdel(eaten_atom)
+	src.visible_message("<span class='warning>[src] eats [eaten_atom]!</span>", "<span class='notice'>You eat [eaten_atom].</span>")
```
Comment (AnturK) [inline code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:376]: Drop src. Also you're using atom you just deleted here.  Not a good idea. better to capture name before that.

File: code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm [hunk 3]
```diff
@@ -299,21 +299,149 @@
...
+
+	if(IS_EDIBLE(potential_food))
+		if(prank_pouch.len >= 8)
+			to_chat(src, "<span class='warning'>Your prank pouch is filled to the brim! You don't think you can swallow any more morsels right now.</span>")
+			return FALSE
+		return TRUE
+
+///This proc eats the atom, certain funny items are stored directly in the prank pouch while bananas grant a heal based on their potency and the peels are retained in the pouch.
+/mob/living/simple_animal/hostile/retaliate/clown/mutant/glutton/proc/eat_atom(atom/movable/eaten_atom)
+	if(istype(eaten_atom, /obj/item/food/pie/cream) || istype(eaten_atom, /obj/item/food/grown/tomato) || istype(eaten_atom, /obj/item/food/meatclown))
```
Comment (AnturK) [inline code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:365]: Would be better cleaner if the allowed objects were a static list.

**Review by AnturK — CHANGES_REQUESTED** [review]
Also eaten objects should probably be removed from the list on exited and/or handle_atom_del implemented for it

File: code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm (hunk 2 above)
Comment (Krysonism) [reply code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:376]: I will put the visible_message further up before anything is deleted.

File: code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm (hunk 3 above)
Comment (Krysonism) [reply code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm]: Done

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-3). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Comment by timothymtorres** [comment]
I used `gitblame` to track the PR that was causing #60933

It seems you forget to set a `maxHealth` var for the glutton and maybe an `active` var isn't getting used properly?

**Pull request merged**